x/warp/contract_warp_handler.go

```diff
@@ -0,0 +1,132 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package warp
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/subnet-evm/precompile/contract"
+	predicateutils "github.com/ava-labs/subnet-evm/utils/predicate"
+	"github.com/ava-labs/subnet-evm/vmerrs"
+	warpPayload "github.com/ava-labs/subnet-evm/warp/payload"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+var (
+	_ messageHandler = addressedPayloadHandler{}
+	_ messageHandler = blockHashHandler{}
+)
+
+var (
+	getVerifiedWarpMessageInvalidOutput   []byte
+	getVerifiedWarpBlockHashInvalidOutput []byte
+)
+
+func init() {
+	res, err := PackGetVerifiedWarpMessageOutput(GetVerifiedWarpMessageOutput{Valid: false})
+	if err != nil {
+		panic(err)
+	}
+	getVerifiedWarpMessageInvalidOutput = res
+
+	res, err = PackGetVerifiedWarpBlockHashOutput(GetVerifiedWarpBlockHashOutput{Valid: false})
+	if err != nil {
+		panic(err)
+	}
+	getVerifiedWarpBlockHashInvalidOutput = res
+}
+
+type messageHandler interface {
+	packFailed() []byte
+	handleMessage(msg *warp.Message) ([]byte, error)
+}
+
+func handleWarpMessage(accessibleState contract.AccessibleState, input []byte, remainingGas uint64, handler messageHandler) ([]byte, uint64, error) {
+	warpIndex, err := UnpackGetVerifiedWarpMessageInput(input)
+	if err != nil {
+		return nil, 0, fmt.Errorf("%w: %s", errInvalidIndexInput, err)
```

Comment:
Follow up on https://github.com/ava-labs/subnet-evm/pull/830#discussion_r1316663749

I meant other errors like that: https://github.com/ava-labs/subnet-evm/blob/warp-stapling-block-context/accounts/abi/bind/precompilebind/precompile_contract_template.go#L243-L245

Reply:
Sounds good, I'll update this for the non `vmerrs.ErrOutOfGas` errors.